examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImReadCommand.kt

```diff
@@ -35,9 +37,86 @@ class PairOnNetworkLongImReadCommand(
       setFailure("read failure")
     }
 
+    fun checkAttributeTlv(attribute: AttributeState): Boolean = 
+      attribute.getTlv().joinToString().equals("""21, 40, 16, 24""").also {
+      if (!it) {
+          logger.log(Level.INFO, "read attribute tlv failure")
+      }
+    };
+
+    fun checkAttributeJson(attribute: AttributeState): Boolean =
+      attribute.getJson().toString().equals("""{"16:BOOL":false}""").also {
+        if (!it) {
+            logger.log(Level.INFO, "read attribute json failure")
+        }
+    };
+
+
+    fun checkEventTlv(event: EventState): Boolean =
+    event.getTlv().joinToString().equals("""21, 53, 0, 36, 0, 1, 24, 24""").also {
+      if (!it) {
+          logger.log(Level.INFO, "read event tlv failure")
+      }
+    };
+
+    fun checkEventJson(event: EventState): Boolean =
+      event.getJson().toString().equals("""{"0:STRUCT":{"0:UINT":1}}""").also {
+        if (!it) {
+            logger.log(Level.INFO, "read event json failure")
+        }
+    };
+
+    fun checkAllAttributeForBasicCluster(cluster: String): Boolean {
+      val expected =
+        """{"16:BOOL":false,""" +
+          """"65531:ARRAY-UINT":[""" +
+          """0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,18,19,20,65528,65529,65531,65532,65533]}"""
```

Comment:
Asserting about a JSON string may be fragile. There's no guaranteed ordering for keys within a JSON object so if the entries are in a different order (but still correct) I wouldn't expect this test to fail?

Consider instead converting to a JSONObject representation and asserting about the key/value contents?

Reply:
in C++ Json/Tlv converter, we have put the ordering requirement for contents inside structure per Evgeni, so we are validating here.